Firestore/core/test/unit/core/query_test.cc

```diff
@@ -917,6 +918,58 @@ TEST(QueryTest, MatchesAllDocuments) {
   EXPECT_FALSE(query.MatchesAllDocuments());
 }
 
+TEST(QueryTest, OrderByForAggregateAndNonAggregate) {
+  auto col = testutil::Query("coll");
+
+  // Build two identical queries
+  auto query1 = col.AddingFilter(testutil::Filter("foo", ">", 1));
+  auto query2 = col.AddingFilter(testutil::Filter("foo", ">", 1));
+
+  // Compute an aggregate and non-aggregate target from the queries
+  auto aggregateTarget = query1.ToAggregateTarget();
+  auto target = query2.ToTarget();
+
+  EXPECT_EQ(aggregateTarget.order_bys().size(), 0);
+
+  ASSERT_EQ(target.order_bys().size(), 2);
+  EXPECT_EQ(target.order_bys()[0].direction(), Direction::Ascending);
+  EXPECT_EQ(target.order_bys()[0].field().CanonicalString(), "foo");
+  EXPECT_EQ(target.order_bys()[1].direction(), Direction::Ascending);
+  EXPECT_EQ(target.order_bys()[1].field().CanonicalString(), "__name__");
+}
+
+TEST(QueryTest, GeneratedOrderBysNotAffectedByPreviouslyMemoizedTargets) {
+  auto col = testutil::Query("coll");
+
+  // Build two identical queries
+  auto query1 = col.AddingFilter(testutil::Filter("foo", ">", 1));
+  auto query2 = col.AddingFilter(testutil::Filter("foo", ">", 1));
+
+  // query1 - first to aggregate target, then to non-aggregate target
+  auto aggregateTarget1 = query1.ToAggregateTarget();
+  auto target1 = query1.ToTarget();
+
+  // query2 - first to aggregate target, then to non-aggregate target
+  auto aggregateTarget2 = query2.ToAggregateTarget();
+  auto target2 = query2.ToTarget();
```

Comment:
query1 and query2 are same here

Reply:
that's intentional, because we need to make sure the order in which `ToAggregateTarget` and `ToTarget` are called do not mutate the Query object differently.

Reply:
The sequence are same here, aggregate call before non-aggregate, while in web it is
```
// query1 - first to aggregate target, then to non-aggregate target
    const aggregateTarget1 = queryToAggregateTarget(query1);
    const target1 = queryToTarget(query1);

    // query2 - first to non-aggregate target, then to aggregate target
    const target2 = queryToTarget(query2);
    const aggregateTarget2 = queryToAggregateTarget(query2);
    ```

Reply:
Thank you for your persistence @cherylEnkidu.

Reply:
This is now fixed.

Reply:
Oops. Good catch @cherylEnkidu